File: src/main/java/com/official/memento/todo/service/usecase/ToDoPrioritizationUseCase.java
package com.official.memento.todo.service.usecase;

import com.official.memento.todo.domain.entity.ToDo;
import com.official.memento.todo.service.command.ToDoPrioritizationCommand;

import java.util.List;

public interface ToDoPrioritizationUseCase {
    List<List<ToDo>> prioritizeWeekly(ToDoPrioritizationCommand command);
    List<ToDo> prioritizeDaily(ToDoPrioritizationCommand command);
}

File: src/main/kotlin/com/official/memento/todo/service/ToDoPrioritizationUseCase.kt
package com.official.memento.todo.service

import com.official.memento.todo.domain.ToDo
import com.official.memento.todo.service.command.ToDoPrioritizationCommand

@FunctionalInterface
interface ToDoPrioritizationUseCase {
    fun prioritizeWeekly (command: ToDoPrioritizationCommand): List<List<ToDo>>
    fun prioritizeDaily (command: ToDoPrioritizationCommand): List<ToDo>
}
